File: app/src/main/java/com/lis/player_java/data/room/model/ArtistDB.java
package com.lis.player_java.data.room.model;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import org.jetbrains.annotations.NotNull;

@Entity
public class ArtistDB {
    @PrimaryKey
    @NotNull
    public String id;
    public Long musicId;
    public String name;
    public String domain;

    ArtistDB(String id, Long musicId, String name, String domain) {
        this.id = id != null ? id : "";
        this.musicId = musicId;
        this.name = name;
        this.domain = domain;
    }

}

File: app/src/main/java/com/lis/player_java/data/room/model/ArtistDB.kt
package com.lis.player_java.data.room.model

import androidx.room.PrimaryKey
import com.lis.player_java.data.room.model.AlbumForMusic
import com.lis.player_java.data.room.model.GenreType
import androidx.room.Embedded
import androidx.room.Entity
import com.lis.player_java.data.room.model.MusicDB
import com.lis.player_java.data.room.model.ArtistDB
import com.lis.player_java.data.room.model.ArtistType

@Entity
data class ArtistDB(
    @PrimaryKey
    val id: String = "",
    val musicId: Long,
    val name: String,
    val domain: String
)